components/atom/switch/src/index.js

```diff
@@ -0,0 +1,179 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import cx from 'classnames'
+import AtomLabel from '../../label/src/index'
+
+const BASE_CLASS = 'sui-AtomSwitch'
+
+const SIZES = {
+  MEDIUM: 'medium',
+  LARGE: 'large'
+}
+
+const TYPES = {
+  TOGGLE: 'toggle',
+  SELECT: 'select',
+  SINGLE: 'single'
+}
+
+class AtomSwitch extends Component {
+  constructor (props) {
+    super(props)
+
+    const {size, disabled, labelLeft, labelRight, type, name, label, labelOptionalText, onToggle} = props
+    this.onToggle = onToggle
+    this.size = size
+    this.type = type
+    this.state = {
+      toggle: false,
+      disabled: disabled,
+      isFocus: false,
+      labelLeft: labelLeft,
+      labelRight: labelRight,
+      name: name,
+      label: label,
+      labelOptionalText: labelOptionalText
+    }
+
+    window.addEventListener('keydown', this.keyBindings.bind(this), true)
+  }
+
+  keyBindings = (event) => {
+    if (!this.state.isFocus) {
+      return
+    }
+
+    if (event.defaultPrevented) {
+      return
+    }
+
+    switch (event.key) {
+      case 'Enter':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+      case ' ':
+        this.toggleSwitch()
+        event.preventDefault()
+        break
+    }
+  }
+
+  toggleSwitch = function () {
+    if (this.state.disabled) {
+      return
+    }
+    if (this.onToggle) {
+      this.onToggle(!this.state.toggle)
+    }
+    this.setState({toggle: !this.state.toggle})
+  }.bind(this)
```

Comment:
You don't have to do this. You can create the function as an instance one.
https://github.com/tc39/proposal-class-fields#field-declarations

Right now you are creating 2 instances of Function each time.

Reply:
do you mean to do this? https://github.com/SUI-Components/sui-components/blob/master/components/collapsible/basic/src/index.js#L15 

otherwise, I am not sure how to use it on a JS event

Reply:
I meant this:

````
class AtomSwitch extends Component {
   myMethod = () => {
      /* YOUR CODE */
   }
}